sound/soc/sof/core.c

```diff
@@ -236,8 +242,8 @@ static int sof_probe(struct platform_device *pdev)
 	/* initialize sof device */
 	sdev->dev = &pdev->dev;
 	sdev->parent = plat_data->dev;
-	if (plat_data->type == SOF_DEVICE_PCI)
-		sdev->pci = container_of(plat_data->dev, struct pci_dev, dev);
+	if dev_is_pci(plat_data->dev)
+		sdev->pci = to_pci_dev(plat_data->dev);
```

Comment:
if this works (?) can we remove definitions for SOF_DEVICE_PCI/ACPI/SPI then?

Reply:
@plbossart it works on glk. let me see if I can remove the macros

Reply:
@plbossart I 've remove the device type enum and it works for pci. But I'm not sure if there are equivalent calls for ACPI/SPI that we could use to determine the type later on.

Reply:
ok let's leave this for now